deploy/template.yaml

```diff
@@ -3004,6 +3101,728 @@ Resources:
       ComparisonOperator: GreaterThanThreshold
       TreatMissingData: notBreaching
 
+  ####################################################################
+  #                                                                  #
+  # Canary Alarms for individual lambdas                             #
+  #                                                                  #
+  ####################################################################
+
+
+  IssueClientAccessTokenFunctionErrorCanaryAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: UseCanaryDeploymentAlarms
+    Properties:
+      ActionsEnabled: false
+      AlarmActions:
+        - !ImportValue alarm-alerts-topic
+      OKActions:
+        - !ImportValue alarm-alerts-topic
+      AlarmDescription: "Error returned from the IssueClientAccessTokenFunction."
+      AlarmName: !Sub ${AWS::StackName}-${Environment}-IssueClientAccessTokenFunction-ErrorCanary
+      MetricName: Errors
+      Dimensions:
+        - Name: Resource
+          Value: !Sub "${AWS::StackName}-IssueClientAccessTokenFunction:live"
+        - Name: FunctionName
+          Value: !Ref IssueClientAccessTokenFunction
+        - Name: ExecutedVersion
+          Value: !GetAtt IssueClientAccessTokenFunction.Version.Version
+      Namespace: AWS/Lambda
+      Statistic: Sum
+      Unit: Count
+      Period: 60
+      EvaluationPeriods: 3
+      Threshold: 1
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
+
+  IssueClientAccessTokenFunction5xxErrorCanaryAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: UseCanaryDeploymentAlarms
+    Properties:
+      ActionsEnabled: false
+      AlarmActions:
+        - !ImportValue alarm-alerts-topic
+      OKActions:
+        - !ImportValue alarm-alerts-topic
+      AlarmDescription: !Sub "IssueClientAccessTokenFunction returning 5xx response."
+      AlarmName: !Sub ${AWS::StackName}-${Environment}-IssueClientAccessTokenFunction-5xxErrorCanary
+      Namespace: AWS/ApiGateway
+      MetricName: 5XXError
+      Dimensions:
+        - Name: ApiName
+          Value: !Sub IPV Core External API Gateway ${Environment}
+        - Name: Method
+          Value: POST
+        - Name: Stage
+          Value: !Ref Environment
+        - Name: Resource
+          Value: /token
+      Statistic: Sum
+      Unit: Count
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 1
```

Comment:
Are these deliberately stricter than the error alarms?

Reply:
Not intentional, there is a baseline error rate so I've modified these to also be in 3x1 minute, same for the state machine `ExecutionsFailed` alarm E.g. this is the sum over 5 min periods in build over the last 4 weeks:

![executions-failed-core-build](https://github.com/user-attachments/assets/f2d54018-0524-4a66-b918-d4dec9907e6c)

Reply:
Thanks - that's a higher baseline than I'd have expected tbh, so maybe something for me to look into!